posts/2021/11/pydata-extensibility-vision.md

```diff
@@ -0,0 +1,196 @@
+
+First two options are very similar in functionality and usability. But the last option, in addition to multiple dispatch, offers granular control using context managers and provides a way to switch backends for the same array type. It's also possible to make the `uarray` dispatcher work without a context manager making the mechanism implicit and similar to other dispatch libraries. Whether we should encourage the implicit registration of backends or not is an open design issue and there's previous discussion on the topic in [SciPy #14266](https://github.com/scipy/scipy/issues/14266). Hameer Abbasi, one of the `uarray` authors, wrote [a blog post](https://labs.quansight.org/blog/2019/07/uarray-update-api-changes-overhead-and-comparison-to-__array_function__/) about the motivation for the `uarray` and how it compares to NumPy's `__array_function__` dispatch mechanism.
+
+## It's already happening
+
+SciPy has GPU and distributed arrays support [on the roadmap](http://scipy.github.io/devdocs/dev/roadmap.html), but adding this support directly to the library would inflate the scope and increase the maintenance burden significantly. scikit-learn has [explicitly stated](https://scikit-learn.org/stable/faq.html#will-you-add-gpu-support) that they cannot afford GPU-specific code in their codebase. A dispatching system is a low-maintenance solution for extending the functionality of the API.
+
+In 2019 Peter Bell added support for backend switching to `scipy.fft` module ([PR #10383](https://github.com/scipy/scipy/pull/10383)) using `uarray`. Now it's possible to tell SciPy to use, for example, CuPy backend, for computing FFT when CuPy's array is passed to functions from `scipy.fft`.
+
+In this recent blog post, Anirudh Dagar explores array interoperability protocols: `__array_function__` (NEP 18), `__array_namespace__` (NEP 47) and `uarray`. The blog post includes a [section comparing the Array API and `uarray`](https://labs.quansight.org/blog/2021/10/array-libraries-interoperability/#protocol_differences) approaches for interoperability.
+
+## Acknowledgment
+
+Adding dispatch support will take significant effort to get implemented across different projects. Fortunately, AMD recognizes the problem and is willing to help to fix it by funding Quansight developers to do the initial work.
+
+AMD's vision for the developer community is open, portable tools that enable the same source code to run on different hardware platforms. ROCm is an example of this, supporting both OpenMP and HIP on multiple architectures. "AMD and Quansight are aligned in this direction. We believe that by working together and working with the Python community, we can achieve an open, portable solution for all Python developers," says Terry Deem, AMD Product Manager for ROCm.
+
+<p align="left">
+    <img
+     alt="AMD logo."
+     src="/images/sponsor/AMD_E_Blk_RGB.png">
+    <i></i>
+</p>
```

Comment:
This is broken in the current state.

**I'm not sure** but I guess this src path can't be used for adding images to a blog. You will probably need to move the image to `/images/2021/11/` and then change the path accordingly to make this render.

Reply:
maybe it is just a typo? The folder is named `sponsors` rather than `sponsor`.

Reply:
yes indeed, fixed now

Reply:
Yes, I earlier tried to fix the path but that's not the issue. You can still try to see, the logo is never rendered. You need to move it into `/images/2021/11/` to use in the blog.

Reply:
No that's not the case, it works as is. Many other blog posts have images in `images/...` rather than in `posts/`.

Reply:
Interesting. Below is a screenshot of where the logo is supposed to be, on the rendered website locally.
![image](https://user-images.githubusercontent.com/23621655/141609834-0564564e-ed8b-4770-ae2d-d96a52098f25.png)

Reply:
Odd. Works for me:

<img width="306" alt="image" src="https://user-images.githubusercontent.com/98330/141610931-7163ee21-3ea2-4f02-875e-665ea6f557f6.png">

Maybe a difference in versions, not sure what's going on here.

This is the same as in many other blog posts, it really should work. Is it broken for you on other posts? There are a ton of images in `images/` and `images/sponsors/`.